### ISSUE TITLE
Training Geo-Adapter does not seem to converge after 18K iterations

### ISSUE THREAD

**Issue by emjay73** [JaceyHuang/Gen3R#10]
<img width="1236" height="351" alt="Image" src="https://github.com/user-attachments/assets/984d8267-cae0-49e0-8422-579c7e2d3d6c" />

I am training the Geo-Adapter from scratch using 4 GPUs, but the loss/performance does not seem to converge even after 18K iterations.

I used the following arguments for the training script:
```
    "--use_deepspeed",
    "--vggt_path",
    "Pretrained/VGGT-1B",
    "--wan_vae_path",
    "./Pretrained/Wan2.1-T2V-1.3B/Wan2.1_VAE.pth",
    "--dataset_path",
    "./DATA/DyGen3R_format/train_videos_dirs.txt",
    "--video_sample_size",
    "560",
    "--video_sample_stride",
    "2",
    "--video_sample_n_frames",
    "49",
    "--train_batch_size",
    "2",
    "--accumulate_grad_batches",
    "12",
    "--dataloader_num_workers",
    "8",
    "--num_train_epochs",
    "300",
    "--learning_rate",
    "1e-05",
```

Is it critical to train using shorter video lengths initially?

**Comment by JaceyHuang** [comment]
Yes — in general, it is important to start with shorter video lengths when training from scratch.

We would kindly suggest first trying shorter clips in the early stage of training, as this often helps stabilize optimization and allows the model to more easily learn local geometric correspondences before handling longer temporal dependencies.

In addition, you may also consider mixing in indoor data during training. From our experience, training with only outdoor data can sometimes lead to less stable convergence, while incorporating indoor scenes tends to improve training stability and overall robustness.

Please let us know if the issue still persists after trying these adjustments.
